src/io/flutter/run/daemon/FlutterAppManager.java

```diff
@@ -282,8 +291,52 @@ private void eventLogMessage(@NotNull LogMessage message, @NotNull FlutterDaemon
 
   private void eventLogMessage(@NotNull AppLog message, @NotNull FlutterDaemonController controller) {
     RunningFlutterApp app = findApp(controller, message.appId);
-    if (app != null && message.log != null) {
-      app.getConsole().print(message.log + "\n", ConsoleViewContentType.NORMAL_OUTPUT);
+    if (app != null) {
+      if (message.progress) {
+        if (message.finished) {
+          clearLatch();
+        } else {
+          // TODO: Is this the correct API to use to get the current project?
+          Project project = ProjectManager.getInstance().getDefaultProject();
```

Comment:
I'd prefer not to have UI code this deep in the process management framework. Maybe FlutterAppState could host the progress updater. It probably has (or can easily have added) a reference to the Project. I admit to having been in a bit of a hurry when doing the console printing, and should have done it that way.